src/Bicep.Core/Analyzers/Linter/Rules/SimplifyJsonNullRule.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.CodeAction;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class SimplifyJsonNullRule : LinterRuleBase
+    {
+        public new const string Code = "simplify-json-null";
+
+        public SimplifyJsonNullRule() : base(
+            code: Code,
+            description: CoreResources.SimplifyJsonNullRuleDescription,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}"))
+        { }
+
+        public override IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model, DiagnosticLevel diagnosticLevel)
+        {
+            var spanFixes = new Dictionary<TextSpan, CodeFix>();
+            var visitor = new Visitor(spanFixes, model);
+            visitor.Visit(model.SourceFile.ProgramSyntax);
+
+            return spanFixes.Select(kvp => CreateFixableDiagnosticForSpan(diagnosticLevel, kvp.Key, kvp.Value));
+        }
+        
+        private sealed class Visitor : AstVisitor
+        {
+            private readonly Dictionary<TextSpan, CodeFix> spanFixes;
+            private readonly SemanticModel model;
+
+            public Visitor(Dictionary<TextSpan, CodeFix> spanFixes, SemanticModel model)
+            {
+                this.spanFixes = spanFixes;
+                this.model = model;
+            }
+
+            public override void VisitObjectPropertySyntax(ObjectPropertySyntax syntax)
```

Comment:
Is there a reason you're not just visiting `FunctionCallSyntax` directly? e.g.:

```csharp
public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
{
     // your logic from line 73 here
}
```

This should be the only overridden method you need in your visitor - the base implementations of `VisitObjectPropertySyntax`, `VisitVariableDeclarationSyntax` etc will take care of recursing down into the `.Value` properties. See [here](https://github.com/Azure/bicep/blob/fa98538c6b5c59c520e580a2280ddb6a458a7992/src/Bicep.Core/Syntax/AstVisitor.cs#L200) for example.

Reply:
updated